autogpt_platform/backend/backend/blocks/smart_decision_maker.py

```diff
@@ -699,22 +704,25 @@ async def run(
             else:
                 expected_args = {arg: {} for arg in tool_args.keys()}
 
-            # Get field mapping from tool definition
+            # Get the sink node ID and field mapping from tool definition
             field_mapping = (
                 tool_def.get("function", {}).get("_field_mapping", {})
                 if tool_def
                 else {}
             )
+            sink_node_id = (
+                tool_def.get("function", {}).get("_sink_node_id") if tool_def else None
+            )
 
             for clean_arg_name in expected_args:
                 # arg_name is now always the cleaned field name (for Anthropic API compliance)
                 # Get the original field name from field mapping for proper emit key generation
                 original_field_name = field_mapping.get(clean_arg_name, clean_arg_name)
                 arg_value = tool_args.get(clean_arg_name)
 
-                sanitized_tool_name = self.cleanup(tool_name)
+                tool_key = sink_node_id if sink_node_id else self.cleanup(tool_name)
```

Comment:
do the fallback even useful here? should we just raise error when the _sink_node_id is not defined?